### PR TITLE
HEC-1483: test only journey starters

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/uk/gov/hmrc/hecapplicantfrontend/testonly/connectors/AuthLoginStubConnector.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      "enrolment[0].taxIdentifier[0].value" -> enrolmentIdentifier.map(_.value),
+      "enrolment[0].state"                  -> loginData.enrolment.map(_.state)
+    ).collect { case (k, Some(v)) => k -> v }.toMap
+  }
+
+  override def login(loginData: LoginData)(implicit hc: HeaderCarrier): EitherT[Future, Error, WSResponse] = {
+    val formData =
+      requestFormBody(loginData).map { case (k, v) => s"${k.urlEncode}=${v.urlEncode}" }.mkString("&")
+    EitherT[Future, Error, WSResponse](
+      ws.url(authLoginStubUrl)
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/hecapplicantfrontend/testonly/connectors/AuthLoginStubConnector.scala:72]: I couldn't get this post with the form url encoded body to work with the http client from http-verbs we use in the production code - I could only get it to work with the `wSClient`. I couldn't get the auth-login-stub service to read the request body properly.

File: test/uk/gov/hmrc/hecapplicantfrontend/testonly/controllers/JourneyStarterControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        checkPageIsDisplayed(
+          controller.journeyStarter(FakeRequest()),
+          "Start a journey",
+          { doc =>
+            testRadioButtonOptions(
+              doc,
+              List(
+                "Individual, no SA UTR",
+                "Individual, SA return found",
+                "Individual, SA notice to file issued (but return not found)",
```
Comment (priyankadhiman3006) [inline test/uk/gov/hmrc/hecapplicantfrontend/testonly/controllers/JourneyStarterControllerSpec.scala:103]: Should it be return found?

Comment (priyankadhiman3006) [inline test/uk/gov/hmrc/hecapplicantfrontend/testonly/controllers/JourneyStarterControllerSpec.scala:103]: should this "but return not found)" be "return found"?

Comment (AndyHWChung) [reply test/uk/gov/hmrc/hecapplicantfrontend/testonly/controllers/JourneyStarterControllerSpec.scala:103]: no, it should be "return not found" - I had originally put "return found" but a return found from the API automatically gives us the "return found" response mentioned in the above journey

File: app/uk/gov/hmrc/hecapplicantfrontend/testonly/models/EitherUtils.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          .orElse((json \ rightFieldName).validate[B].map(Right(_)))
+
+      override def writes(o: Either[A, B]): JsValue =
+        o.fold(
+          a ⇒ JsObject(Seq(leftFieldName → Json.toJson(a))),
+          b ⇒ JsObject(Seq(rightFieldName → Json.toJson(b)))
+        )
+    }
+
+}
```
Comment (meghana-viswanath) [inline app/uk/gov/hmrc/hecapplicantfrontend/testonly/models/EitherUtils.scala:49]: where is this used?

**Review by meghana-viswanath — APPROVED** [review]

File: app/uk/gov/hmrc/hecapplicantfrontend/testonly/models/EitherUtils.scala (hunk 3 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/hecapplicantfrontend/testonly/models/EitherUtils.scala:49]: It's format for verifier: either[CRN, DOB ] i think

**Review by priyankadhiman3006 — APPROVED** [review]

File: app/uk/gov/hmrc/hecapplicantfrontend/testonly/models/EitherUtils.scala (hunk 3 above)
Comment (AndyHWChung) [reply app/uk/gov/hmrc/hecapplicantfrontend/testonly/models/EitherUtils.scala:49]: yeah, it's for that verifier field for the `SaveTaxCheckRequest` case class for the BE test only endpoint

**Pull request merged**